lib/admin.js

```diff
@@ -239,23 +241,23 @@ define.classMethod('ping', {callback: true, promise:true});
  * @param {string} [password] The password.
  * @param {Admin~resultCallback} [callback] The command result callback
  * @return {Promise} returns Promise if no callback passed
+ * @deprecated This method will no longer be available in the next major release 3.x as MongoDB 3.6 will only allow auth against users in the admin db and will no longer allow multiple credentials on a socket. Please authenticate using MongoClient.connect with auth credentials.
  */
 Admin.prototype.authenticate = function(username, password, options, callback) {
-  var self = this;
-  if(typeof options == 'function') callback = options, options = {};
-  options = shallowClone(options);
-  options.authdb = 'admin';
-
-  // Execute using callback
-  if(typeof callback == 'function') return this.s.db.authenticate(username, password, options, callback);
+  console.warn("Admin.prototype.authenticate method will no longer be available in the next major release 3.x as MongoDB 3.6 will only allow auth against users in the admin db and will no longer allow multiple credentials on a socket. Please authenticate using MongoClient.connect with auth credentials.");
+  var finalArguments = [this.s.db];
+  if(typeof username == 'string') finalArguments.push(username);
+  if(typeof password == 'string') finalArguments.push(password);
+  if(typeof options == 'function') { 
+    finalArguments.push({ authdb: 'admin' });
+    finalArguments.push(options);
+  } else {
+    finalArguments.push(assign({}, options, { authdb: 'admin' }));
```

Comment:
authdb should always be admin so it should be the last one overwriting any user option.